homeassistant/components/qvrpro/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""Support for QVR Pro NVR software by QNAP."""
+
+import logging
+
+from pyqvrpro import Client
+from pyqvrpro.client import AuthenticationError, InsufficientPermissionsError
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+from .const import CONF_EXCLUDE_CHANNELS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_EXCLUDE_CHANNELS, default=[]): vol.All(
+                    cv.ensure_list_csv, [cv.positive_int]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the QVR Pro component."""
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = config[DOMAIN][CONF_HOST]
+    excluded_channels = config[DOMAIN][CONF_EXCLUDE_CHANNELS]
+
+    try:
+        qvrpro = Client(user, password, host)
+
+        channel_resp = qvrpro.get_channel_list()
+
+    except InsufficientPermissionsError:
+        _LOGGER.error("User must have Surveillance Management permission")
+        return False
+    except AuthenticationError:
+        _LOGGER.error("Authentication failed")
+        return False
+
+    channels = []
+
+    for channel in channel_resp["channels"]:
+        if channel["channel_index"] + 1 in excluded_channels:
+            continue
+
+        channels.append(QVRChannel(**channel))
+
+    hass.data[DOMAIN] = {"channels": channels, "client": qvrpro}
+
+    load_platform(hass, "camera", DOMAIN, {}, config)
```

Comment:
Please import and rename the constant DOMAIN from camera and use it here.

Reply:
Making this change results in the QVR camera component not being loaded at all.  I thought domain was shared across all components in an integration.  Only other thing I can think of, is that you wanted me to import the QVR domain from the QVR camera component, but that seems unnecessary since both components source that from the integration constants.  Can you clarify what you are requesting here?

Reply:
Sure, what I was requesting was to import the camera domain e.g.
https://github.com/home-assistant/home-assistant/blob/6f8f23238ad2db3acfc949cb497dca33dd604ab4/homeassistant/components/camera/const.py#L2
from [camera/const.py](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/camera/const.py) so you can use it in line 63. However, it needs to be renamed to something else to avoid name clashing e.g. `CAMERA_DOMAIN`:

Reply:
AH... I see.  You want me to replace the string. I thought you were asking me to replace the domain.  Too many DOMAIN constants floating around and my brain didn't want to separate them.

Reply:
No worries I should have been clearer with my request. You are not the only one, mixing these up happens very often.